File: bubble_sort.c
#include<stdio.h>
void bubble_sort(int n,int arr[n]);
void main(){
    int n;
    printf("enter size of array :");
    scanf("%d",&n);
    int arr[n];
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }
     
    bubble_sort(n,arr);
    printf("Sorted array: ");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }

}
void bubble_sort(int n,int arr[n]){
for(int i=0;i<n;i++){
    for(int j=0;j<n-1-i;j++){
    if(arr[j]>arr[j+1]){
          int temp=arr[j];
          arr[j]=arr[j+1];
          arr[j+1]=temp;
    }
    }

}
}

File: Datastructures/BubbleSort.java
public class BubbleSort {
    
    static void sort(int[] arr){
      for(int i=0;i<arr.length;i++){
        for(int j=0;j<arr.length-1-i;j++){
            if(arr[j]>arr[j+1]){
                int temp=arr[j];
                arr[j]=arr[j+1];
                arr[j+1]=temp;
            }
        }
      }
    }
    public static void main(String args[]){
       int[] arr={3,2,5,4,1};
       System.out.print("Sorted array : ");
       sort(arr);
       for(int i=0;i<arr.length;i++){
        System.out.print(arr[i]+" ");
       }
    }
}
